### DISCUSSION TITLE
No name

### DISCUSSION THREAD

**Discussion by hartenthaler** [fisharebest/gedcom-name#12] [General]
If there is a person and no name is known for this person, then there should be no NAME tag for this individual.

As far as I interpret the GEDCOM standard, the following records are invalid:

> 1 NAME

> 1 NAME //

**Comment by Norwegian-Sardines** [comment]
Should the “no name” designation be restricted to just the `NAME` tag, or should subtags also have some designation that a name part is not know.  For example: a given name is known but a surname is not.  Some application have suggested to their users to enter 5 “_” (1 NAME John /_____/),and others suggest a “?” or some other value in place of name parts that are unknown.

**Reply by hartenthaler** [reply]
For me there is a general rule: if you don't know it leave it empty. If there is no source for an event (maybe the death of a person that lived 200 years ago) then I will not generated a source "unknown" for the "1 DEAT Y" event. 

If I don't know the first name it will be "1 NAME /Meyer/", if I don't know the surname and know that there is one existing it will be "1 NAME John //". And "2 GIVN John", but no "2 SURN".

For me this better than using ?, _, NN, x, or something else. If your tree has only Chinese names, it would look strange to use N.N. as an unknown name. 

Maybe when presenting a name in a report it would be necessary to indicate an unknown name part (using ... or ___), but this implementation of a genealogical program and not part of the GEDCOM standard.

**Reply by Norwegian-Sardines** [reply]
My question is more of a statement that some software companies want their users to enter data like underscores or question marks when a name part is unknown.  It is my belief that this is wrong and that GEDCOM should provide an indication that a name part as well as the entire name is unknown.

**Reply by Tim-Partridge** [reply]
I think we should avoid putting in any text like "No name" to represent a name being unknown. By contrast, in Mongolia some people are actually called "Nergüi [No name]" or "Medekhgüi [I don't know]" to confuse evil spirits. Wikipedia's Mongolian name article has a discussion of this and similar names in the Taboo names section.

**Comment by hartenthaler** [comment]
> GEDCOM should provide an indication that a name part as well as the entire name is unknown.

For me, this indication is that NAME is not there (entire name is unknown) or some subtags are not there (depends if subtags are in a future release mandatory or not).

**Reply by Norwegian-Sardines** [reply]
The v7 of GEDCOM contain a NON_EVENT_STRUCTURE, Indicates that a specific type of event, given in the payload, did not happen.

And my point is that should `NAME` tag also be included to indicate that the whole name is not known!

An extension of that would be to indicate that, should a subtag (name part) also be given that indication.

What I’ve seen in GEDCOMs in the wild is all kinds of other indications like question marks, 5 underscores and other words all indicated as correct by the software provider.

I also have issue currently with software where they can’t show a difference between individuals that a surname is unknown and where an individual has no surname.  

Surname unknown
1 NAME John //
vs.
No surname
1 NAME John

**Comment by Norwegian-Sardines** [comment]
In v5.5.1 of the GEDCOM specification on page 55 the following example was indicated for NAME_PERSONAL:

Examples:
William Lee (given name only or surname not known)

This is contrary to what I wrote above and what my software does for persons with an unknown name.
